pkg/client/client_test.go

```diff
@@ -2647,6 +2671,33 @@ U5wwSivyi7vmegHKmblOzNVKA5qPO8zWzqBC
 				Expect(deps.Items[1].Name).To(Equal(dep4.Name))
 			})
 
+			It("should not deep copy results when UnsafeDisableDeepCopy is set", func() {
```

Comment:
This is already tested, no need to add another: https://github.com/kubernetes-sigs/controller-runtime/blob/71f7db556ca57ce7ea6563f77d739f0d2a54233a/pkg/cache/cache_test.go#L2465

Reply:
To me, it's a little odd that this is in cache_test.go when it's a List option on the client? Does this test belong there?